.github/workflows/main.yml

```diff
@@ -0,0 +1,52 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+    push:
+        branches:
+            - master
+            - 'dev/**'
+    pull_request:
+        branches:
+            - master
+            - 'dev/**'
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+    # This workflow contains a single job called "build"
+    build:
+        # The type of runner that the job will run on
+        runs-on: ubuntu-latest
+
+        strategy:
+            matrix:
+                node-version: [16.x]
+
+        # Steps represent a sequence of tasks that will be executed as part of the job
+        steps:
+            # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+            - uses: actions/checkout@v2
+
+            - name: Run a one-line script
+              uses: actions/setup-node@v2
+              with:
+                  node-version: ${{ matrix.node-version }}
+
+            - run: npm install --prefix ./test-app/backend
+
+            - run: npm run lint-check --prefix ./test-app/backend
+
+            - run: npm run lint-fix --prefix ./test-app/backend
+
+            - run: npm run test --prefix ./test-app/backend
+
+            - run: npm install --prefix ./test-app/frontend
+
+            - run: npm run lint-check --prefix ./test-app/frontend
+
+            - run: npm run lint-fix --prefix ./test-app/frontend
+
+            - run: npm run test --prefix ./test-app/frontend
```

Comment:
Instead of running all of these separately, can you make a script in each `package.json` that does the install, lint-check, lint-fix, and test. Like `npm run quality-checks`